### ISSUE TITLE
Suffix transforms (-ej, -ist) don't create semantic clusters

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#148]
## Summary

The suffix transform tests for -ej (places) and -ist (professions) fail to create expected semantic clusters.

## Test Results

```
TestAffixTransformations::test_ej_places_cluster FAILED
  -ej places should cluster: mean_sim=-0.023 (expected > 0.3)

TestAffixTransformations::test_ist_professions_cluster FAILED
  -ist professions should cluster: mean_sim=0.038 (expected > 0.3)
```

## Analysis

The suffix transforms are working (they modify embeddings) but they're not creating the expected "place" or "profession" semantic clusters. This is likely because:

1. **Training objective mismatch**: Current affix training focuses on:
   - Contrastive learning (distinguish affixed from non-affixed)
   - Anti-collapse (prevent all affixed words from being identical)
   
2. **Missing category signal**: There's no explicit training signal that says "all -ej words should be similar to each other because they're places"

### Option A: Category-aware affix training (recommended)
Add a training objective that encourages:
- `lernejo` similar to `laborejo` (both places)
- `artisto` similar to `muzikisto` (both professions)

### Option B: Add suffix cluster loss
```python
# Pseudo-code
for suffix in ['ej', 'ist', 'an', ...]:
    affixed_words = get_words_with_suffix(suffix)
    embeddings = [model.embed(w) for w in affixed_words]
    cluster_loss = -mean_pairwise_similarity(embeddings)
    total_loss += cluster_loss
```

### Option C: Accept current behavior
The current transforms may be sufficient for RAG retrieval even without tight clustering.

## Current Workaround

Mark tests as expected failures until Stage 2 (suffix clustering) training is implemented.

## Related
- Part of Stage 1 suffix training improvements
- Consider for Stage 2 if clustering is needed for reasoning

## Labels
- `training`
- `enhancement`
